Fix coprocess deadlock

krader1961 opened the pull request:
Fixes #563

siteshwar commented on src/cmd/ksh93/sh/jobs.c:
```diff
@@ -48,6 +48,8 @@
 
 #define NJOB_SAVELIST 4
 
+//volatile bool vmbusy_flag = false;
```
Shall this be removed ?

krader1961 commented:
I refactored this so the wrappers are in a new *src/lib/libast/misc/vmbusy.c* module and added more comments and a TODO about refactoring the `job_waitsafe()` function so that the job reaping is never done in interrupt context.
